qiskit/synthesis/linear/linear_depth_lnn.py

```diff
@@ -222,8 +222,8 @@ def _optimize_cx_circ_depth_5n_line(mat):
 
     # According to [1] the synthesis is done on the inverse matrix
     # so the matrix mat is inverted at this step
-    mat_inv = mat.copy()
-    mat_cpy = calc_inverse_matrix(mat_inv)
+    mat_inv = mat.copy().astype(bool)
+    mat_cpy = calc_inverse_matrix(mat_inv).astype(bool)
```

Comment:
The return from rust is already of dtype bool, so you shouldn't need to cast anything. Do you need a copy here, if so I'd do this rust side instead. You can do that with `to_pyarray()` instead of `into_pyarray()`. Either way you can remove the `astype()` call here:

```suggestion
    mat_cpy = calc_inverse_matrix(mat_inv)
```